engine/class_modules/sc_shaman.cpp

```diff
@@ -8536,6 +8635,26 @@ void shaman_t::create_buffs()
                             ->set_default_value_from_effect_type( A_ADD_PCT_MODIFIER, P_TICK_TIME )
                             ->apply_affecting_conduit( conduit.call_of_flame );
 
+  buff.fireheart = make_buff( this, "fireheart", spell.t28_2pc_ele )
+                       // TODO: confirm if duration is affected by conduit and tier set 4pc bonus
+                       ->set_duration( buff.fire_elemental->buff_duration() )
+                       ->set_tick_callback( [ this ]( buff_t* b, int, timespan_t ) {
+                         if ( buff.lava_surge->check() )
```

Comment:
Could consider making a shaman_t accessor for handling the Lava Surge, instead of replicating code here and in flame_shock_t?

Reply:
Done